public/app/features/explore/PrometheusListView/RawListContainer.tsx

```diff
@@ -0,0 +1,162 @@
+
+import { DataFrame, Field } from '@grafana/data/src';
+import { Button, stylesFactory } from '@grafana/ui/src';
+
+import {
+  getRawPrometheusListItemsFromDataFrame,
+  RawPrometheusListItemEmptyValue,
+} from '../utils/getRawPrometheusListItemsFromDataFrame';
+
+import { ItemLabels } from './ItemLabels';
+import RawListItem from './RawListItem';
+
+export type instantQueryRawVirtualizedListData = { Value: string; __name__: string; [index: string]: string };
+
+export interface RawListContainerProps {
+  tableResult: DataFrame;
+}
+
+const getRawListContainerStyles = stylesFactory(() => {
+  return {
+    wrapper: css`
+      height: 100%;
+      overflow: scroll;
+    `,
+    mobileWrapper: css`
+      height: 100%;
+      overflow: scroll;
+    `,
+    header: css`
+      display: flex;
+      justify-content: space-between;
+      align-items: center;
+      padding: 10px 0;
+    `,
+  };
+});
+
+const mobileWidthThreshold = 480;
+const numberOfColumnsBeforeExpandedViewIsDefault = 2;
+
+/**
+ * The container that provides the virtualized list to the child components
+ * @param props
+ * @constructor
+ */
+const RawListContainer = (props: RawListContainerProps) => {
+  const { tableResult } = props;
+  let dataFrame = cloneDeep(tableResult);
+  const styles = getRawListContainerStyles();
+  let listRef = useRef<List | null>(null);
+
+  const valueLabels = dataFrame.fields.filter((field) => field.name.includes('Value'));
+  const items = getRawPrometheusListItemsFromDataFrame(dataFrame);
+  const { width } = useWindowSize();
+  const [isExpandedView, setIsExpandedView] = useState(
+    width <= mobileWidthThreshold || valueLabels.length > numberOfColumnsBeforeExpandedViewIsDefault
+  );
+
+  const onContentClick = () => {
+    setIsExpandedView(!isExpandedView);
+  };
+
+  useEffect(() => {
+    // After the expanded view has updated, tell the list to re-render
+    listRef.current?.resetAfterIndex(0, true);
+  }, [isExpandedView]);
+
+  const calculateInitialHeight = (length: number): number => {
+    const maxListHeight = 600;
+    const shortListLength = 10;
+
+    if (length < shortListLength) {
+      let sum = 0;
+      for (let i = 0; i < length; i++) {
+        sum += getListItemHeight(i, true);
+      }
+
+      return Math.min(maxListHeight, sum);
+    }
+
+    return maxListHeight;
+  };
+
+  const getListItemHeight = (itemIndex: number, isExpandedView: boolean) => {
+    const singleLineHeight = 32;
+    const additionalLineHeight = 22;
+    if (!isExpandedView) {
+      return singleLineHeight;
+    }
+    const item = items[itemIndex];
+
+    // Height of 1.5 lines, plus the number of non-value attributes times the height of additional lines
+    return 1.5 * singleLineHeight + (Object.keys(item).length - valueLabels.length) * additionalLineHeight;
+  };
+
+  return (
+    <section>
+      <header className={styles.header}>
+        <Button
+          variant="secondary"
+          title="Add query"
```

Comment:
is this  the title we actually want for the button?

another thing regarding this button is that it lacks some accessibility requirements. i'm not 100% sure the expanded state here is actually expaning/hiding anything, but maybe we should consider it as a toggle button if it is toggling some behaviour: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/button_role#associated_aria_roles_states_and_properties

Reply:
Good catch with the button title, that was just a sloppy copy/paste error. I'm not sure what else to add for the screen reader markup, as this button only changes the display property (block/inline) of the span elements. I added role=button, but let me know if there's something else we can do to help improve the accessibility UX.

Reply:
i see the button has now aria-hidden=true, this makes it practically inaccessible for screen readers users, we should remove it.

also, `role="button"` is implicit for buttons.

my suggestion was mostly using `aria-pressed` to indicate screen readers this is a toggle button:
![Screenshot 2022-12-30 at 10 12 47](https://user-images.githubusercontent.com/1170767/210059273-d5f7f534-c837-4995-b989-89ab1349f68e.png)


the issue wit that is that changing content makes screen readers (voice over in this case) skip reading the new state:

https://user-images.githubusercontent.com/1170767/210060183-9a64eee6-e472-466f-b645-d30089e4422e.mov


while preserving the content between renders and only use the icon as an indicator of its state for non-screen reader users works nicely (but is less than ideal):
https://user-images.githubusercontent.com/1170767/210060086-7b902dc4-c4c7-4622-80a5-87fbe126401e.mov


Another option (probably the best), which is also more in line with what is being done in the logs panel for instance, is to use a switch:
https://user-images.githubusercontent.com/1170767/210060728-ccc28be8-54c7-4ec1-92d7-ebb7b8952d50.mov

```tsx
        <Field label="Expand results">
          <Switch value={isExpandedView} onChange={onContentClick} id="someid" />
        </Field>
```